src/foundry/client/core/issues.d.mts

```diff
@@ -0,0 +1,196 @@
+     */
+    #usabilityIssues: Record<string, ClientIssues.UsabilityIssue>;
+
+    /** The minimum supported resolution. */
+    static #MIN_RESOLUTION: { WIDTH: 1024; HEIGHT: 700 };
+
+    /**
+     * The minimum supported client versions.
+     */
+    static #BROWSER_TESTS: {
+      /**
+       * @defaultValue
+       * ```typescript
+       * {
+       *   minimum: 24;
+       *   match: RegExp;
+       *   message: "ERROR.ElectronVersion";
+       * }
+       * ```
+       */
+      Electron: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 92;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Chromium: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 90;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Firefox: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 15.4;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Safari: ClientIssues.BROWSER_TEST;
+    };
+
+    /**
+     * Add a Document to the count of module-provided sub-types.
+     * @param documentName      - The Document name.
+     * @param subType           - The Document's sub-type.
+     * @param options.decrement - Decrement the count rather than incrementing it. default: `false`
+     */
+    #countDocumentSubType(documentName: DocumentType, subType: string, options?: { decrement?: boolean }): void;
+
+    /**
+     * Detect the user's browser and display a notification if it is below the minimum required version.
+     */
+    #detectBrowserVersion(): void;
+
+    /**
+     * Record a reference to a resolution notification ID so that we can remove it if the problem is remedied.
+     */
+    #resolutionTooLowNotification: number;
+
+    /**
+     * Detect the user's resolution and display a notification if it is too small.
+     */
+    #detectResolution(): void;
+
+    /**
+     * Detect and display warnings for known performance issues which may occur due to the user's hardware or browser
+     * configuration.
+     * @internal
+     */
+    _detectWebGLIssues(): void;
+
+    /**
+     * Add an invalid Document to the module-provided sub-type counts.
+     * @param documentName      - The Document name.
+     * @param source            - The Document's source data.
+     * @param options.decrement - Decrement the counter rather than incrementing it. default: `false`
+     * @internal
+     *
+     * @remark official documentation says "invalid", but the document can be valid, it is simply not yet validated.
+     */
+    _countDocumentSubType(documentName: DocumentType, source: unknown, options?: { decrement?: boolean }): void;
```

Comment:
protected, remove internal

Reply:
Used in client-document & document-collection